### PR TITLE
Add Present Value calculation for FRA.

### LINKED ISSUE #23: Fra Exploration
This issue is to document the exploration of FRA's in the QuantSA library.

**FRA Present Value Calculation:**
Implement FRA_PV (valuation via Monte Carlo).

To-do on the FRA pv branch:
- [x] Search CoreExtensions for any reusable logic or interfaces for pricing.
- [x]  Write the Monte Carlo FRA valuation.
- [x] Add FRA_PV to ProductPVs.
- [x] Commit frequently with descriptive messages.
- [ ] Push and link PR to the issue once done.

### PR DESCRIPTION
**Pull request by rwx-oberholzer-marcelle**
✅ **Pull Request Checklist**

### 🔍 Self-Review

* [x] Destination branch is correct.
* [x] Code has been self-reviewed and is scoped to the PR’s purpose.
* [x] No debugging code, print statements, or unrelated changes.
* [x] Code is clean, well-commented, and follows project standards.
* [x] Documentation is updated, if applicable.

### 🧪 Testing & Build

* [x] All relevant tests (unit/integration) pass locally.
* [x] Edge cases and error handling are considered.
* [x] No new warnings or errors in the build or test output.

### 📄 Clarity

* [x] PR title and description are clear and descriptive.
* [x] Purpose and context of the change are well explained.
* [x] Related issues are referenced (e.g., `Fixes #123`).

### 👥 Collaboration

* [x] Reviewers assigned and feedback requested.
* [x] All review comments have been addressed or responded to.

### CODE REVIEW

**Review by rwx-jacquier-robert — CHANGES_REQUESTED** [review]
Great, just be sure to add a test for it. You should try and find an independent example for fra values that you can use for the expected value.

File: QuantSA/QuantSA.CoreExtensions/ProductPVs/Rates/FraPVs.cs [hunk 1]
```diff
@@ -0,0 +1,18 @@
...
+{
+    public static class FraPVs
+    {
+        /// <summary>
+        /// Values the FRA by discounting the generated cashflow.
+        /// </summary>
+        public static double CurvePv(this FRA fra, IDiscountingSource discountCurve)
+        {
+            var cfs = fra.GetCFs();  // internally uses _nearDate, _rate, _accrualFraction etc.
+            return cfs.PV(discountCurve);  // extension method in QuantSA
```
Comment (rwx-jacquier-robert) [inline QuantSA/QuantSA.CoreExtensions/ProductPVs/Rates/FraPVs.cs]: Unnecessary comments.

File: QuantSA/QuantSA.CoreExtensions.Test/ProductPVs/Rates/FRAPVsTests.cs [hunk 2]
```diff
@@ -0,0 +1,79 @@
...
+
+            public Currency GetCurrency() => _currency;
+
+            public double GetDF(Date date)
+            {
+                var t = (date - _anchorDate) / 365.0;
+                return Math.Exp(-_rate * t);
+            }
+
+            public bool CanBeA<T>(MarketDataDescription<T> description, IMarketDataContainer container) where T : class, IMarketDataSource => false;
```
Comment (rwx-jacquier-robert) [inline QuantSA/QuantSA.CoreExtensions.Test/ProductPVs/Rates/FRAPVsTests.cs]: Remove this, is ununsed.

**Review by rwx-jacquier-robert — CHANGES_REQUESTED** [review]

**Review by rwx-jacquier-robert — APPROVED** [review]

**Comment by rwx-jacquier-robert** [comment]
@MarcelleOberholzer no need to re-request review when it was already approved :)

**Review by rwx-jacquier-robert — APPROVED** [review]
